### ISSUE TITLE
Crash with drones

### ISSUE THREAD

**Issue by Darkere** [TeamPneumatic/pnc-repressurized#633]
### Minecraft Version
1.15
### Forge Version
1.15.2-31.2.36
### Mod Version
pneumaticcraft-repressurized-1.15.2-1.5.0-73.jar
### Describe your problem, including steps to reproduce it
Not a clue on how to reproduce it. 
https://gist.github.com/Darkere/6d18b9a6e16d920a3db9451030916b1e

Mainly concerned about what I could do to get the server back online :/

### Any other comments?
This might actually be related to another issue reported around the same time the server started crashing. 
https://github.com/NillerMedDild/Enigmatica5/issues/526

**Comment by desht** [comment]
I don't think it's related to that other issue, but it did look familiar, and I remembered this got reported in 1.12.2 as well - #399.  I added a workaround there (could never trace why it was crashing... read the issue), but I stopped it crashing completely.

Looks like the "fix" didn't make it in 1.15, so I'll get it in now.

**Comment by desht** [comment]
OK, build 78 from https://jenkins.k-4u.nl/job/PneumaticCraft-Repressurized-1.15/ should get you up and running again.  I'll be getting a 1.5.1 release out fairly soon.

**Comment by Darkere** [comment]
Build 78 seems to have loaded up just fine. Thanks for the super fast response :)

**Comment by Darkere** [comment]
OR NOT LOL

**Comment by Darkere** [comment]
Looks to me like the same issue? 
https://gist.github.com/Darkere/bbf456b4bb524867df4a32d18e1b6b2d

**Comment by desht** [comment]
Hmm.  Only possibility now is that there are more than 36 itemstacks in this order... wtf?  I can add a workaround for this too, but no clue how it's gotten into this state.  Could you upload your `config/pneumaticcraft/AmadronPlayerOffers.cfg` file to somewhere I can take a look at it?

**Comment by Darkere** [comment]
I was just about to do that. 
https://gist.github.com/Darkere/b21fdc0fd16138bef0fce77530ff5f00

**Comment by desht** [comment]
Can't see anything obviously wrong there, but when you start up with build 79 (available now), check for a "retrieveOrderItems: unexpected size" log message in the server log.  Will be interested to see which order is triggering that...

**Comment by desht** [comment]
Only order that looks possibly suspicious is:
```
{
      "id": "pneumaticcraft:mutetiefling_1602415143",
      "input": {
        "type": "ITEM",
        "id": "minecraft:emerald",
        "amount": 16
      },
      "output": {
        "type": "ITEM",
        "id": "simplefarming:fish_sandwich",
        "amount": 64
      },
```
If those fish sandwiches don't stack (do they?) that would end up with a 64-long list of items.  The 36 limit is in place since that's the max size of a drone's inventory, with the full 35 inventory upgrades.

**Comment by Darkere** [comment]
[m←[1;31m[13:17:08] [Server thread/ERROR] [pneumaticcraft/]: retrieveOrderItems: unexpected size (37) for itemstack list for offer 197 x 12 plastic @ DimensionType{minecraft:overworld} BlockPos{x=681, y=65, z=864}

![image](https://user-images.githubusercontent.com/4283717/95981951-c792e500-0e1f-11eb-90b6-ef597790df5f.png)

**Comment by MuteTiefling** [comment]
Ooops! 

I had tossed in about 2 inventories worth of plastic into a diamond chest for the amadron to come and pick up. Not even buying anything

**Comment by desht** [comment]
Yeah... that's it ;(  2364 plastic = 37 stacks.

I need to handle that better, clearly...

**Comment by desht** [comment]
Right... build 80 should fix it *properly*.  The Amadrone should no longer attempt to collect more than 36 stacks a time when restocking player offers.  So any excess will be left in the chest, and collected on the next restocking run (30 seconds later IIRC).

**Comment by desht** [comment]
Fixed in 1.5.1 release
(and 2.4.3 for MC 1.16.3)
